services/src/main/java/org/fao/geonet/api/records/MetadataInsertDeleteApi.java

```diff
@@ -145,6 +142,33 @@
     private final String API_PARAM_BACKUP_FIRST = "Backup first the record as MEF in the metadata removed folder.";
     private final String API_PARAM_RECORD_TYPE = "The type of record.";
 
+    @Autowired
```

Comment:
In current master, you "can't" use Autowired because of the multinode mode (in fact you can but it may not work for some users). So first we have to remove that first. We agreed on that last year, the work is done in here https://github.com/geonetwork/core-geonetwork/pull/3475 but has been waiting for review since weeks/months. As discussed Thursday with @josegar74 and @PascalLike we will merge it next week. 

So I'm for this change but it has been done for the whole API package in #3475 after dropping multinode mode which then allows to use Autowired with no risk ... here you're duplicating work. I may be a bit rough here but we should take more attention to pending PRs and probably discuss the review process in Bolsena.

Reply:
Was looking at 3475, it impact a lot of files, but for MetadataInsertDeleteApi, seems not to be complete, e.g. https://github.com/geonetwork/core-geonetwork/pull/3475/files#diff-0a151ba101d3ace43356caba460e533fL196.

Should I rebase the request on  fxprunayre:feature/multinode-replaced-by-subportal ?

Reply:
Was looking at 3475, it impact a lot of files, but for MetadataInsertDeleteApi, seems not to be complete, e.g. https://github.com/geonetwork/core-geonetwork/pull/3475/files#diff-0a151ba101d3ace43356caba460e533fL196.

Should I rebase the request on  fxprunayre:feature/multinode-replaced-by-subportal ?